### PR TITLE
Introduce Pattern fusing with Hardware config support for ONNX

### PR DESCRIPTION
**Pull request by kshpv**
### Changes

Add support of Hardware patterns fusing; Add support of Hardware config

### Reason for changes

65600

### Related tickets

75427

### Tests

Added test on number mapped metatypes in NNCFGraphs; The full test coverage will be added with the support of the quantization algorithm

### CODE REVIEW

**Comment by FyodorK** [comment]
Jenkins please retry a build

**Comment by FyodorK** [comment]
Jenkins please stop all builds

**Comment by FyodorK** [comment]
Jenkins please stop all builds

**Comment by FyodorK** [comment]
Jenkins please retry a build

**Comment by FyodorK** [comment]
Jenkins please retry a build

File: nncf/experimental/onnx/graph/pattern_operations.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+from nncf.common.graph.patterns import merge_two_types_of_operations
+
+LINEAR_OPERATIONS = {'type': ['Conv',
```
Comment (alexsu52) [inline nncf/experimental/onnx/graph/pattern_operations.py:16]: Could you use meta types here?

Comment (kshpv) [reply nncf/experimental/onnx/graph/pattern_operations.py:16]: We have a special ticket on implementing pattern matching algorithm based on metatypes- 58797.

File: tests/onnx/test_onnx_metatypes.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+from nncf.experimental.onnx.graph.metatypes.onnx_ops import ConcatLayerMetatype
+from nncf.experimental.onnx.graph.metatypes.onnx_ops import AddLayerMetatype
+
+from tests.onnx.models import LinearModel
+from tests.onnx.models import MultiInputOutputModel
+from tests.onnx.models import ModelWithIntEdges
+
+TEST_MODELS = [LinearModel, MultiInputOutputModel, ModelWithIntEdges]
+REF_METATYPES_COUNTERS = [
+    [InputNoopMetatype, ConvolutionMetatype, BatchNormMetatype, ReluMetatype, GlobalAveragePoolMetatype,
```
Comment (ljaljushkin) [inline tests/onnx/test_onnx_metatypes.py:22]: How does it work now with metatype, if you only plan to support metatypes in next PR (ticket 58797)?

File: tests/onnx/test_onnx_metatypes.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+@pytest.mark.parametrize(("model_creator_func, ref_metatypes"),
+                         zip(TEST_MODELS, REF_METATYPES_COUNTERS))
+def test_mapping_onnx_metatypes(model_creator_func, ref_metatypes):
+    model = model_creator_func()
+    nncf_graph = GraphConverter.create_nncf_graph(model.onnx_model)
+    ref_metatypes_counter = Counter(ref_metatypes)
+    metatypes_counter = Counter()
+    for node in nncf_graph.get_all_nodes():
+        metatypes_counter[node.metatype] += 1
+    assert ref_metatypes_counter == metatypes_counter
```
Comment (ljaljushkin) [inline tests/onnx/test_onnx_metatypes.py:38]: I'd simplify this code:
```suggestion
    actual_metatypes = [node.metatype for node in nncf_graph.get_all_nodes()]
    assert Counter(ref_metatypes) == Counter(actual_metatypes)
```

File: nncf/experimental/onnx/graph/patterns.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+ limitations under the License.
+"""
+
+from nncf.common.graph.patterns import GraphPattern
+
+
+def create_h_sigmoid_act() -> GraphPattern:
+    pattern = GraphPattern()
+
+    input_pattern_node = pattern.add_node(label='*INPUT_NODE*', type=GraphPattern.NON_PATTERN_NODE_TYPE)
```
Comment (ljaljushkin) [inline nncf/experimental/onnx/graph/patterns.py:20]: why is it with asterisks?

**Review by vshampor — APPROVED** [review]

**Review by ljaljushkin — APPROVED** [review]

**Review by alexsu52 — APPROVED** [review]

File: nncf/experimental/onnx/graph/patterns.py (hunk 4 above)
Comment (kshpv) [reply nncf/experimental/onnx/graph/patterns.py:20]: The 'label' field is just for the visualization of patterns. We have an unspoken rule, that the special nodes are designated with asterisks.

**Pull request merged**